core/src/WebhookDispatcher/ValueObject/DispatchWebhookRequest.php

```diff
@@ -174,9 +190,10 @@ public static function createFromRequest(array $bodyValues, array $headerValues)
             (string) $bodyValues['category'],
             $bodyValues['summary'] ?? null,
             $bodyValues['orderId'] ?? null,
-            (string) $headerValues['Shop-Id'],
-            (string) $headerValues['Merchant-Id'],
-            (string) $headerValues['Psx-Id']
+            (string) $headerValues['Shop_Id'],
+            (string) $headerValues['Merchant_Id'],
+            (string) $headerValues['Psx_Id'],
```

Comment:
Actually, the received headers are formatted like that, at the validation step.
https://github.com/PrestaShopCorp/ps_checkout/discussions/1383#discussioncomment-14706811
```suggestion
          (string) $headerValues['shopId'],
          (string) $headerValues['merchantId'],
          (string) $headerValues['firebaseId'],
```

Reply:
https://github.com/PrestaShopCorp/ps_checkout/blob/v5.0.5/core/src/WebhookDispatcher/Provider/WebhookHeaderProvider.php